src/core/mtu_discovery.c

```diff
@@ -0,0 +1,238 @@
+/*++
+
+    Copyright (c) Microsoft Corporation.
+    Licensed under the MIT License.
+
+Abstract:
+
+    This module handles the MTU discovery logic.
+
+    Upon a new path being validated, MTU discovery is started
```

Comment:
nit: on this and the header, can you please align comments to 80 characters as much as possible (but not more)?